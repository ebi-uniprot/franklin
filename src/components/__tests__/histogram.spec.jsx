import { render } from '@testing-library/react';
import { Histogram } from '..';

describe('HistogramFilter component', () => {
  it('should render when nBins prop is provided and with the correct number of bins', async () => {
    const { asFragment, findByTestId, getAllByTestId } = render(
      <Histogram values={[...Array(10).keys()]} nBins={2} />
    );
    await findByTestId('histogram');
    expect(asFragment()).toMatchSnapshot();
    expect(getAllByTestId('histogram-bar')).toHaveLength(2);
  });

  it('should render when binSize prop is provided and with the correct number of bins', async () => {
    const { asFragment, findByTestId, getAllByTestId } = render(
      <Histogram values={[...Array(10).keys()]} binSize={2} />
    );
    await findByTestId('histogram');
    expect(asFragment()).toMatchSnapshot();
    expect(getAllByTestId('histogram-bar')).toHaveLength(5);
  });

  it('should shade bins that are in range', async () => {
    const { findByTestId, getAllByTestId } = render(
      <Histogram
        values={[...Array(10).keys()]}
        binSize={2}
        selectedRange={[3, 7]}
      />
    );
    await findByTestId('histogram');
    const bars = getAllByTestId('histogram-bar');
    expect(
      bars.filter(({ className }) =>
        className.includes('histogram__bar--within-range')
      )
    ).toHaveLength(3);
  });
});
