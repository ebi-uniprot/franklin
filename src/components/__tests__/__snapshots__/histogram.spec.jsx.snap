// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistogramFilter component should render when binSize prop is provided and with the correct number of bins 1`] = `
<DocumentFragment>
  <div
    class="histogram"
    data-testid="histogram"
  >
    <div
      class="histogram__bar-container"
      style="height: 300px;"
    >
      <div
        class="histogram histogram__bar histogram__bar--within-range"
        data-testid="histogram-bar"
        style="transform: scaleY(0.4); transition-delay: 0ms;"
      />
      <div
        class="histogram histogram__bar histogram__bar--within-range"
        data-testid="histogram-bar"
        style="transform: scaleY(0.4); transition-delay: 100ms;"
      />
      <div
        class="histogram histogram__bar histogram__bar--within-range"
        data-testid="histogram-bar"
        style="transform: scaleY(0.4); transition-delay: 200ms;"
      />
      <div
        class="histogram histogram__bar histogram__bar--within-range"
        data-testid="histogram-bar"
        style="transform: scaleY(0.4); transition-delay: 300ms;"
      />
      <div
        class="histogram histogram__bar histogram__bar--within-range"
        data-testid="histogram-bar"
        style="transform: scaleY(0.4); transition-delay: 400ms;"
      />
    </div>
    <svg
      height="80"
      style="top: 300px;"
      width="0"
    />
    <svg
      class="y-axis"
      height="300"
      width="80"
    >
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(50, 0)"
      >
        <path
          class="domain"
          d="M0.5,300.5V0.5"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,300.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            0.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,270.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            0.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,240.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            1.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,210.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            1.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,180.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            2.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,150.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            2.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,120.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            3.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,90.50000000000001)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            3.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,60.499999999999986)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            4.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,30.499999999999993)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            4.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            5.0
          </text>
        </g>
      </g>
    </svg>
  </div>
</DocumentFragment>
`;

exports[`HistogramFilter component should render when nBins prop is provided and with the correct number of bins 1`] = `
<DocumentFragment>
  <div
    class="histogram"
    data-testid="histogram"
  >
    <div
      class="histogram__bar-container"
      style="height: 300px;"
    >
      <div
        class="histogram histogram__bar histogram__bar--within-range"
        data-testid="histogram-bar"
        style="transform: scaleY(1); transition-delay: 0ms;"
      />
      <div
        class="histogram histogram__bar histogram__bar--within-range"
        data-testid="histogram-bar"
        style="transform: scaleY(1); transition-delay: 250ms;"
      />
    </div>
    <svg
      height="80"
      style="top: 300px;"
      width="0"
    />
    <svg
      class="y-axis"
      height="300"
      width="80"
    >
      <g
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(50, 0)"
      >
        <path
          class="domain"
          d="M0.5,300.5V0.5"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,300.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            0.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,270.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            0.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,240.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            1.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,210.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            1.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,180.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            2.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,150.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            2.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,120.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            3.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,90.50000000000001)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            3.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,60.499999999999986)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            4.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,30.499999999999993)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            4.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.5)"
        >
          <line
            stroke="currentColor"
            x2="0"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            5.0
          </text>
        </g>
      </g>
    </svg>
  </div>
</DocumentFragment>
`;
